homework2.py

```diff
@@ -0,0 +1,428 @@
+		2. if this path is a directory
+		3. if this path is a file
+	if input path doesn't exist, return False and empty list
+
+	Arguments:
+	path: input path as a string
+
+	Returns:
+	exist_flag: True if path exist, False otherwise
+	path_info_list: list containing 3 boolean varaibles for 3 checks
+		performed if path exists, return empty list if path doesn't
+		exist.
+		''''
+def check_path(xyz):
+    list123=[]
+    if path.exists(xyz) == True:
+        list123.append(path.isabs(xyz))
+        list123.append(path.isfile(xyz))
+        list123.append(path.isdir(xyz))
+    print(str(path.exists(xyz)) + ',' + str (list123))
+
+	check_path('/Users/sun/Desktop/ds-class-intro')
+
+	'''
+	Example:
+	check_path('some/path/of/a/directory')
+	# if it exists
+	>>> True, [False, True, False]
+
+	# if doesn't exist
+	>>> False, []
+	'''
+
+
+
+'''       ****************************Question Mark***********************************
+2.def read_csv(file):
+	''''''
+	reads in a csv file then return the total number of lines in it
+
+	Arguments:
+	file: path to the file to read
+
+	Returns:
+	Number of rows in the input file
+
+	Example:
+	read_csv('AMZN.csv')
+	>>> 14
+	'''
+
+'''There is something wrong with this question, but I cannot figure out	'''
+def read_csv1(file):
+    open(file,'r')
+    reader = csv.reader(file)
+    num_line = len(list(reader))
+    print(num_line)
+read_csv1('AMZN.csv')
+
+def read_csv2(file):
+    open(file,'r')
+    num_line=0
+    reader = csv.reader(file)
+    for row in reader:
+        num_line+=1
+    print(num_line)
+read_csv2('AMZN.csv')
+
+def read_csv3(file):
+    open(file,'r')
+    reader = csv.reader(file)
+    row_count = sum(1 for row in reader)
+    print(row_count)
+read_csv2('AMZN.csv')
+
+'''
+3.def write_csv(data_list, output_file):
+	''' '''
+	write out a csv file for the data list (structed as list of list),
+	where each item in the data_list is a row in output file, and
+	every element in the sublist is separated by comma
+
+	Arguments:
+	data_list: input data list, each elemet is a list representing
+		a row with values for each column
+	file: path to save the csv file
+
+	Returns:
+	None
+
+	Example:
+	data_list = [(1,2,3,4), (5,6,7,8), (9,10,11,12)]
+	write_csv(data_list, 'example.csv')
+
+	'example.csv' looks like below:
+	1,2,3,4
+	5,6,7,8
+	9,10,11,12
+	''''
+''''
+data_list = [(1,2,3,4), (5,6,7,8), (9,10,11,12)]
+def write_csv(data_input,file_output):
+	with open('example.csv', 'w', newline='') as file:
+    	writer = csv.writer(file)
+    	writer.writerow([input('row 1 column 1:'), input('row 1 column 2:'), input('row 1 column 3:'), input('row 1 column 4:')]
+		writer.writerow([input('row 2 column 1:'), input('row 2 column 2:'), input('row 2 column 3:'), input('row 2 column 4:')])
+		writer.writerow([input('row 3 column 1:'), input('row 3 column 2:'), input('row 3 column 3:'), input('row 3 column 4:')])
+'''
+
+'''need a better solution'''
+def write_csv(data_input,file_output):
```

Comment:
this is what we called "hard-coding": your code will break once input is changed. your function should be modified to fix this and write as many rows as there are in `data_list`